### PR TITLE
Remove x-ms based correlation.

### PR DESCRIPTION
**Pull request by cijothomas**
Fix Issue #939  .
Also made minor improvements to unit tests as well.

- [ ] I ran Unit Tests locally.
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue if available.

For significant contributions please make sure you have completed the following items:

- [ ] Changes in public surface reviewed
- [ ] Design discussion issue #
- [ ] The PR will trigger build, unit tests, and functional tests automatically. If your PR was submitted from fork - mention one of committers to initiate the build for you.
	  If you want to to re-run the build/tests, the easiest way is to simply Close and Re-Open this same PR. (Just click 'close pull request' followed by 'open pull request' buttons at the bottom of the PR)

- Please follow [these] (https://github.com/Microsoft/ApplicationInsights-aspnetcore/blob/develop/Readme.md) instructions to build and test locally.

### CODE REVIEW

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 1]
```diff
@@ -615,7 +582,8 @@ private void EndRequest(HttpContext httpContext, long timestamp)
 
                 this.client.TrackRequest(telemetry);
 
-                var activity = httpContext?.Features.Get<Activity>();
+                // Stop what we started.
+                var activity = Activity.Current;
```
Comment (cijothomas) [inline src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: This was an unrelated fix. Activity was never stored into httpContext.Features, so we never stopped the activity we started. Good code hygiene.

File: test/Microsoft.ApplicationInsights.AspNetCore.Tests/HostingDiagnosticListenerTest.cs [hunk 2]
```diff
@@ -133,6 +137,35 @@ public void TestSdkVersionIsPopulatedByMiddleware(bool isAspNetCore2)
             Assert.Contains(SdkVersionTestUtils.VersionPrefix, requestTelemetry.Context.GetInternalContext().SdkVersion);
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void TestSdkVersionIsPopulatedByMiddleware(bool isAspNetCore2)
```
Comment (cijothomas) [inline test/Microsoft.ApplicationInsights.AspNetCore.Tests/HostingDiagnosticListenerTest.cs]: leaving the name as before.. the name is clearly not indicative of what test is doing. Will address in separte PR.

File: src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 1 above)
Comment (Dmitry-Matveev) [reply src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: Could it be that framework store it for us? I do not know, so I would've guessed it may.

Comment (cijothomas) [reply src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs]: Framework (hosting) don't store it. SDK creates and starts activity with the name "ActivityCreatedByHostingDiagnosticListener", and framework has no knowledge of it.
